catkin_ws/src/planner/src/substates/utility/controller.py

```diff
@@ -169,35 +169,35 @@ def rotateEuler(self,ang):
         if z is None: z = self.theta_z
         self.rotate(euler_to_quaternion(x,y,z))
 
-    def state(self,pos,ang):
-        x,y,z = pos
-        if any(x is None for x in ang) and any(x is not None for x in ang):
-            raise ValueError("Invalid state goal: quaternion cannot have a combination of None and valid values. Goal received: {}".format(ang))
-        w,wx,wy,wz = ang
-        goal_state = self.get_state_goal([x,y,z,w,wx,wy,wz],do_not_displace)
-        self.StateQuaternionStateClient.send_goal_and_wait(goal_state)
-    
-    def stateDelta(self,pos,ang):
+
+    def state(self, pos, ang, isEuler=False, isDelta=False):
+        if isEuler:
+            wx,wy,wz = ang
+
+            if not isDelta:
```

Comment:
I don't think these two ifs need to be nested

Reply:
@MrMondrian ?

Reply:
I think as a general programming practice the less nesting the better. This code doesn't need two ifs, one would do but how it would it be slower?

Reply:
It does need two if's tho. If its a euler angle, convert it to quaternion. When you convert to quaternion, you need to know if its a delta or not to know what to replace None's with. So either way you put it you need two if's when there are eulers and quaternions combined. Maybe I'm missing something?

Reply:
yeah you're right my bad